brave/adblock-resources
# Fix youtube theater mode fix script

Turns out that [session cookies ARE actually persisted across browser restarts](https://github.com/brave/brave-browser/issues/28379#issuecomment-1640677254) on MacOS, which is where I was testing this fix, and why we didn't catch this. I suspect that I had a pending browser update which is why in our initial testing I was able to repro the issue :(

The issue is basically that we need to reload the page if we come across the problematic scenario i.e. session cookie was wiped and we have a localstorage backup, otherwise YT won't enable theater mode.

TESTING:
1. Get the `base64` output of the script resource being tested
2. Put that in `"content"` for `"brave-youtube-theater-fix"` in `resources.json` at `~/Library/Application Support/BraveSoftware/Brave-Browser/mfddibmblmbccpadfndgakiopmmhebop/<latestversion>`
3. Restart Brave (Release)
4. Go to Youtube and select theater mode
5. Close tab, go to `brave://settings/cookies/detail?site=youtube.com` and delete the `wide` cookie. This now replicates the scenario that we are trying to protect against i.e. the user selected theater mode and then closed the tab and the `wide` cookie is deleted.
6. Open YT again. Check if it opens in theater mode.
